patroni/api.py

```diff
@@ -89,8 +91,20 @@ def do_GET(self, write_status_code_only=False):
         patroni = self.server.patroni
         cluster = patroni.dcs.cluster
 
-        replica_status_code = 200 if not patroni.noloadbalance and \
-            response.get('role') == 'replica' and response.get('state') == 'running' else 503
+        replica_status_code = 503
+        if response.get('role') == 'replica' and response.get('state') == 'running' \
+           and not patroni.noloadbalance:
+            max_replica_lag = self.path_query.get('lag') and parse_int(self.path_query.get('lag')[0], 'B') \
+                                                         or sys.maxsize
+            leader_optime = cluster and cluster.last_leader_operation or 0
+            replayed_location = response.get('xlog').get('replayed_location') or 0
+            current_lag = leader_optime - replayed_location
+            if current_lag >= 0:
+                max_replica_lag = max(max_replica_lag, 0)
```

Comment:
That will set a negative `max_replica_lag` to 0. What will it bring us?

Reply:
you suggested change includes the fix. thanks